pkg/controller/machine/machine_suite_test.go

```diff
@@ -22,33 +22,33 @@ import (
 	"github.com/kubernetes-incubator/apiserver-builder/pkg/test"
 
 	"sigs.k8s.io/cluster-api/pkg/apis"
-	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1/testutil"
 	"sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset"
 	"sigs.k8s.io/cluster-api/pkg/controller/sharedinformers"
 	"sigs.k8s.io/cluster-api/pkg/openapi"
 )
 
-type fakeActuator struct{}
-
-func (fakeActuator) Create(*clusterv1.Cluster, *clusterv1.Machine) error           { return nil }
-func (fakeActuator) Delete(*clusterv1.Machine) error                               { return nil }
-func (fakeActuator) Update(c *clusterv1.Cluster, machine *clusterv1.Machine) error { return nil }
-func (fakeActuator) Exists(*clusterv1.Machine) (bool, error)                       { return false, nil }
-
 func TestMachine(t *testing.T) {
 	testenv := test.NewTestEnvironment()
 	config := testenv.Start(apis.GetAllApiBuilders(), openapi.GetOpenAPIDefinitions)
 	cs := clientset.NewForConfigOrDie(config)
 
 	shutdown := make(chan struct{})
 	si := sharedinformers.NewSharedInformers(config, shutdown)
-	controller := NewMachineController(config, si, fakeActuator{})
-	controller.Run(shutdown)
-
-	t.Run("machineControllerReconcile", func(t *testing.T) {
-		machineControllerReconcile(t, cs, controller)
-	})
-
+	actuator := &TestActuator{}
+	controller := NewMachineController(config, si, actuator)
+	controller.RunAsync(shutdown)
+
+	// Create a cluster object.
+	cluster := testutil.GetVanillaCluster()
+	cluster.Name = "cluster-1"
+	if _, err := cs.ClusterV1alpha1().Clusters("default").Create(&cluster); err != nil {
```

Comment:
Why is this here? Why don't the sub tests create the clusters if they need for their specific test?

Reply:
It's test case for machine resource, I think it's just common setup so that we can create machine.